### ISSUE TITLE
Computer available memory is reduced from 14G to 100M until the program is killed

### ISSUE THREAD

**Issue by zhangqingbo1** [neka-nat/image_completion_keras#6]
Hello,I trained the cmp_model Tc = 18 times and d_model Td=2 times, then after the all_module trained 40 times ,the program was killed with the following error, and I find my computer available memory is reduced from 14G to 100M until the program is killed:
terminate called after throwing an instance of 'std::bad_alloc'
    what():  std::bad_alloc
Abandoned (core dumped)

**Comment by zhangqingbo1** [comment]
I checked the computer memory usage and find that the memory increased dramatically in the third step of training(n > Tc+Td)，which is the following code. I think the n >= tc + td loop has been creating models continuously  and it's taking up memory.
                if n >= tc + td:
                    d_container.trainable = False
                    cropping.arguments = {'x1': points[:, 0], 'y1': points[:, 1],
                                          'x2': points[:, 2], 'y2': points[:, 3]}
                    all_model = Model([org_img, mask],
                                      [cmp_out, d_container([cmp_out, cropping(cmp_out)])])
                    all_model.compile(loss=['mse', 'binary_crossentropy'],
                                      loss_weights=[1.0, alpha], optimizer=optimizer)
                    g_loss = all_model.train_on_batch([inputs, masks],
                                                      [inputs, valid])

**Comment by neka-nat** [comment]
I updated master branch.
Please try it with tensorflow1.1.0 and keras2.0.4.
Thank you.

**Comment by zhangqingbo1** [comment]
It works well, thank you so much, you have helped me a lot !
